### PR TITLE
avm2: Add a basic ContextMenu stub

### PR DESCRIPTION
**Pull request by adrian17**
Just enough for this to work (which is what a lot of content ends up doing):
```
mc.contextMenu = new ContextMenu();
mc.contextMenu.hideBuiltInItems();
```

### CODE REVIEW

File: core/src/avm2/globals.rs [hunk 1]
```diff
@@ -837,6 +851,14 @@ pub fn load_player_globals<'gc>(
         script
     );
 
+    // package `flash.ui`
```
Comment (torokati44) [inline core/src/avm2/globals.rs:854]: Copy-paste bug in this comment! (:

**Comment by ousia** [comment]
@adrian17,

if `.hideBuiltInItems();` also hides _Enter fullscreen_, I would ask that #5137 would be reviewed and merged too.

I mean, the code is already there and it is important to have the functionality where it is needed.

Many thanks for your help,

**Comment by adrian17** [comment]
> if .hideBuiltInItems(); also hides Enter fullscreen

AFAIK it doesn't. It only hides the "builtin items" native to Flash, the ones controlled with https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/ContextMenuBuiltInItems.html . Ruffle-specific items like `About` and `Fullscreen` are not affected, and that PR doesn't affect them either.

I thought #5137 was waiting @kmeisthax 's review tbh. I'll be unavailable for this weekend, but could review it afterwards if no one else picks it up before me.

File: core/src/avm2/globals/flash/ui/contextmenu.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+) -> Result<Value<'gc>, Error> {
+    Ok(Value::Undefined)
+}
+
+fn hide_built_in_items<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    // TODO: replace this by a proper implementation.
```
Comment (kmeisthax) [inline core/src/avm2/globals/flash/ui/contextmenu.rs:37]: Since this is a stub method, we should `log::warn` when this is called.

**Review by kmeisthax — CHANGES_REQUESTED** [review]

File: core/src/avm2/globals/flash/ui/contextmenu.rs (hunk 2 above)
Comment (adrian17) [reply core/src/avm2/globals/flash/ui/contextmenu.rs:37]: This particular method isn't really a stub, it's just a simplified implementation; in practice it'll appear to do what one expects it to do as long as they stick to the "simplest path". And if they do anything more complex, we'll still get it logged due to missing other properties, right?

Comment (adrian17) [reply core/src/avm2/globals/flash/ui/contextmenu.rs:38]: Done :P

**Pull request merged**